### PR TITLE
AwsSession copies `endpoint_url` if region is the same

### PR DESCRIPTION
**Pull request by yitchen-tim**
*Description of changes:*
If the `region` in `aws_session.copy_session(region)` is the same as the one of `aws_session`, the endpoint url is also copied.

Example:
```
import boto3
from braket.devices import Devices
from braket.aws import AwsSession, AwsDevice

region_name = "us-west-2"
endpoint_url = "some-endpoint"
braket_client = boto3.client("braket", region_name=region_name, endpoint_url=endpoint_url)
aws_session = AwsSession(braket_client=braket_client)

device = AwsDevice(Devices.Amazon.SV1, aws_session=aws_session)
task_batch = device.run_batch([Circuit().x(0)]*2, shots=100)
```
is expected to create tasks in `some-endpoint`. But currently it would create tasks in the default endpoint of users' region. This PR fix this issue and creates task in `some-endpoint`.

*Testing done:*
tox

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/amazon-braket/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/amazon-braket/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md#commit-your-change)
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/amazon-braket/amazon-braket-sdk-python/blob/main/README.md) and [API docs](https://github.com/amazon-braket/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md#documentation-guidelines) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have checked that my tests are not configured for a specific region or account (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/braket/aws/aws_session.py [hunk 1]
```diff
@@ -824,8 +824,18 @@ def copy_session(
                 region_name=new_region,
                 profile_name=profile_name,
             )
+        new_braket_client = (
+            boto_session.client(
+                "braket", region_name=new_region, endpoint_url=self.braket_client.meta._endpoint_url
```
Comment (shpface) [inline src/braket/aws/aws_session.py:829]: Can we do this without depending on the internal `_endpoint_url` of the boto client?

**Comment by rmshaffer** [comment]
Would you be able to update the PR description with an example of the user problem that this fix solves?

File: src/braket/aws/aws_session.py (hunk 1 above)
Comment (yitchen-tim) [reply src/braket/aws/aws_session.py:829]: The `endpoint_url` is only defined in Braket's boto client. If we don't want to depend `endpoint_url` on boto client, I think we could add `endpoint_url` as an attribute of AwsSession in addition to boto client.

**Comment by yitchen-tim** [comment]
> @yitchen-tim Would you be able to update the PR description with an example of the user problem that this fix solves?

yes, I added an generic example to the description. A more specific example is in an internal tracking board.

File: src/braket/aws/aws_session.py (hunk 1 above)
Comment (yitchen-tim) [reply src/braket/aws/aws_session.py:829]: @shpface Any thought on this?

**Comment by ajberdy** [comment]
Can we solve this by using environment variables?

```
import os
from braket.devices import Devices
from braket.aws import AwsSession, AwsDevice

os.environ["AWS_REGION"] = "us-west-2"
os.environ["BRAKET_ENDPOINT"] = "some-endpoint"

device = AwsDevice(Devices.Amazon.SV1)
task_batch = device.run_batch([Circuit().x(0)]*2, shots=100)
```

I believe should work without any code changes

**Comment by yitchen-tim** [comment]
thank you, this `os.environ` works well for me

**Pull request closed without merging**